lib/elixir/lib/exception.ex

```diff
@@ -1997,11 +1997,11 @@ defmodule Protocol.UndefinedError do
 
   @impl true
   def message(%{protocol: protocol, value: value, description: description}) do
-    "protocol #{inspect(protocol)} not implemented for #{inspect(value)} of type " <>
+    "protocol #{inspect(protocol)} not implemented for #{inspect(value, pretty: true)} of type " <>
       value_type(value) <> maybe_description(description) <> maybe_available(protocol)
   end
 
-  defp value_type(%{__struct__: struct}), do: "#{inspect(struct)} (a struct)"
+  defp value_type(%{__struct__: struct}), do: "#{inspect(struct, pretty: true)} (a struct)"
```

Comment:
This is just an atom, no need for pretty:

```suggestion
  defp value_type(%{__struct__: struct}), do: "#{inspect(struct)} (a struct)"
```